pkg/pull/pull.go

```diff
@@ -310,7 +310,31 @@ func Pull(upstreamURI string, pullOptions PullOptions) (string, error) {
 		IsReadOnly: pullOptions.RewriteImageOptions.IsReadOnly,
 	}
 
-	needsConfig, err := kotsadmconfig.NeedsConfiguration(pullOptions.AppSlug, pullOptions.AppSequence, pullOptions.AirgapRoot != "", kotsKinds, registrySettings)
+	renderOptions := base.RenderOptions{
+		SplitMultiDocYAML: true,
+		Namespace:         pullOptions.Namespace,
+		RegistrySettings:  registrySettings,
+		ExcludeKotsKinds:  pullOptions.ExcludeKotsKinds,
+		Log:               log,
+		AppSlug:           pullOptions.AppSlug,
+		Sequence:          pullOptions.AppSequence,
+		IsAirgap:          pullOptions.AirgapRoot != "",
+	}
+	log.ActionWithSpinner("Creating base")
+	io.WriteString(pullOptions.ReportWriter, "Creating base\n")
+
+	commonBase, helmBases, renderedKotsKindsMap, err := base.RenderUpstream(u, &renderOptions)
+	if err != nil {
+		log.FinishSpinnerWithError()
+		return "", errors.Wrap(err, "failed to render upstream")
+	}
+
+	renderedKotsKinds, err := kotsutil.KotsKindsFromMap(renderedKotsKindsMap)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to load rendered kotskinds from map")
+	}
+
+	needsConfig, err := kotsadmconfig.NeedsConfiguration(pullOptions.AppSlug, pullOptions.AppSequence, pullOptions.AirgapRoot != "", renderedKotsKinds, registrySettings)
```

Comment:
I believe this happened before attempting to render base because of things like this https://app.testim.io/#/project/wpYAooUimFDgQxY73r17/branch/use-rendered-kots-kinds/test/8lupz8KBYa9SGzDO/step/mFwAJVtdRncnSncN/viewer/screenshots?result-id=QSkV5cWC3frZ0dZI&path=s2IRYvVg6O8W5IWX%3AmFwAJVtdRncnSncN